bundles/org.openhab.binding.mikrotik/src/main/java/org/openhab/binding/mikrotik/internal/handler/MikrotikBaseThingHandler.java

```diff
@@ -0,0 +1,226 @@
+                    refreshChannel(channelUID);
+                } else {
+                    try {
+                        executeCommand(channelUID, command);
+                    } catch (Exception e) {
+                        logger.warn("Unexpected error handling command = {} for channel = {} : {}", command, channelUID,
+                                e.getMessage());
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing {}!", getThing().getUID());
+        cancelRefreshJob();
+        if (getVerifiedBridgeHandler() == null) {
+            updateStatus(OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "This thing requires a RouterOS bridge");
+            return;
+        }
+
+        Class<?> klass = (Class<?>) (((ParameterizedType) getClass().getGenericSuperclass())
+                .getActualTypeArguments()[0]);
+        this.config = (C) getConfigAs(klass);
+        logger.trace("Config for for {} ({}) is {}", getThing().getUID(), getThing().getStatus(), config);
+
+        if (!config.isValid()) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, String.format("%s is invalid", klass.getSimpleName()));
+            return;
+        }
+
+        updateStatus(ONLINE);
+        logger.debug("Finished initializing {}!", getThing().getUID());
+    }
+
+    @Override
+    protected void updateStatus(ThingStatus status, ThingStatusDetail statusDetail, @Nullable String description) {
+        logger.trace("Updating {} status to {} / {}", getThing().getUID(), status, statusDetail);
+        if (status == ONLINE || (status == OFFLINE && statusDetail == ThingStatusDetail.COMMUNICATION_ERROR)) {
+            scheduleRefreshJob();
+        } else if (status == OFFLINE
+                && (statusDetail == ThingStatusDetail.CONFIGURATION_ERROR || statusDetail == ThingStatusDetail.GONE)) {
+            cancelRefreshJob();
+        }
+
+        // update the status only if it's changed
+        ThingStatusInfo statusInfo = ThingStatusInfoBuilder.create(status, statusDetail).withDescription(description)
+                .build();
+        if (!statusInfo.equals(getThing().getStatusInfo())) {
+            super.updateStatus(status, statusDetail, description);
+        }
+    }
+
+    private void scheduleRefreshJob() {
+        synchronized (this) {
+            if (refreshJob == null) {
+                int refreshPeriod = getVerifiedBridgeHandler().getBridgeConfig().refresh;
+                logger.debug("Scheduling refresh job every {}s", refreshPeriod);
+                refreshJob = scheduler.scheduleWithFixedDelay(this::scheduledRun, refreshPeriod, refreshPeriod,
+                        TimeUnit.SECONDS);
+            }
+        }
+    }
+
+    private void cancelRefreshJob() {
+        synchronized (this) {
+            if (refreshJob != null) {
+                logger.debug("Cancelling refresh job");
```

Comment:
Can this message be replaced by using the debugger? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
Hmm, this is more for being able to diagnose potential problems for other users after asking to increase the log level for this binding.
Afaik, the default logging level is INFO, so no debug messages will be logged. Or am I wrong?

Reply:
Yes, but it's eating up some resources, though. Fair enough. You can keep it.